### PR TITLE
Added a new Mutation: diffUpdate${type}

### PR DESCRIPTION
**Pull request by bigfish24**
Uses deep-object-diff library: https://github.com/mattphillips/deep-object-diff to check if the input has any changes. However, it lazily applies the entire input if there is any change compared to selectively applying changes.

Will provide basic diffing to prevent reapplying the same changes over and over. cc @ianpward

### CODE REVIEW

File: src/service.ts [hunk 1]
```diff
@@ -492,6 +494,38 @@ export class GraphQLService {
...
+      const oldRealmObject = context.realm.objectForPrimaryKey(type, pkValue);
+      const oldObject = JSON.parse(JSON.stringify(oldRealmObject));
+
+      let result = diff(oldObject, newObject);
+      result = JSON.parse(JSON.stringify(result)); // Not sure why this needs parsed?
+
+      // TODO: apply each change selectively
+      if (Object.keys(result).length !== 0) {
+        context.realm.write(() => {
+          result = context.realm.create(type, newObject, true);
```
Comment (simonask) [inline src/service.ts]: Hm, I'm not super familiar with how this stuff works, but this doesn't seem right? At least it seems like it will always create new objects rather than update fields on existing objects.

Comment (nirinchev) [reply src/service.ts]: it passes `true` as the last argument, which would mean `addOrUpdate`

Comment (simonask) [reply src/service.ts]: Thanks.

We avoid boolean parameters religiously in our code for this reason. ;-)

Anyway, does this mean that the `TODO` above is wrong?

Comment (nirinchev) [reply src/service.ts]: I'm guessing the TODO refers to setting only changed properties - the current implementation will upsert the entire object (generating a set for each property). Using the diff, we can only set the ones that have actual changes.

File: src/service.ts [hunk 2]
```diff
@@ -492,6 +494,38 @@ export class GraphQLService {
...
+    mutationResolver[`diffUpdate${type}`] = (_, args, context) => {
+      this.validateWrite(context);
+
+      const schema = context.realm.schema;
+      const objectSchema = this.objectSchemaWithName(schema, type);
+      const pkName = objectSchema.primaryKey;
+      const newObject = args.input;
+      const pkValue = newObject[pkName];
+      const oldRealmObject = context.realm.objectForPrimaryKey(type, pkValue);
+      const oldObject = JSON.parse(JSON.stringify(oldRealmObject));
```
Comment (nirinchev) [inline src/service.ts]: Why is this needed? When doing tests locally, I could generate proper diff without any of the copying.

File: src/service.ts (hunk 1 above)
Comment (bigfish24) [reply src/service.ts]: Yes assumption is the bulk of extra operations today are doing upserts against unchanged objects, and not that upsetting an object that has 1 field changed will be too many operations compared to just changing that one field.

File: src/service.ts (hunk 2 above)
Comment (bigfish24) [reply src/service.ts]: I didn’t know if the library can diff a Realm object vs a deep copy of the object making it a plain object.

Comment (nirinchev) [reply src/service.ts]: Okay, I'll test some more and if I don't find any issues, I'll remove it.

File: src/service.ts (hunk 1 above)
Comment (simonask) [reply src/service.ts]: This could be insufficient, if for instance the objects contain an "updatedAt" field that is frequently changed without many other changes to the object.
